src/Commands/Workflow/WaitCommand.php

```diff
@@ -175,69 +182,170 @@ protected function waitForWorkflow(
-    //     $target_commit,
-    //     $maxWaitInSeconds = 180,
-    //     $maxNotFoundAttempts = null
-    // ) {
-    //     $wfl = null;
-    //     $wflc = $site->getWorkflowLogs();
-    //     if (!$wflc instanceof WorkflowLogsCollection) {
-    //         throw new TerminusException(
-    //             'Workflow logs could not be retrieved for site: {site}',
-    //             ['site' => $site->id,]
-    //         );
-    //     }
-
-    //     // Remove workflows that are not for the environment $env_name.
-    //     $wflc->filterForEnvironment($env_name);
-
-    //     // Find the latest workflow that matches the commit hash
-    //     $wfl = $wflc->findLatestFromOptionsArray([
-    //         'target_commit' => $target_commit,
-    //     ]);
-
-    //     $current_time = time();
-    //     if ($maxWaitInSeconds > 0) {
-    //         $end_time = $current_time + $maxWaitInSeconds;
-    //     } else {
-    //         $end_time = 0;
-    //     }
-
-    //     // If we didn't find a workflow, then we need to wait for one to be created
-    //     if (!$wfl instanceof WorkflowLog) {
-    //         // sleep to give the workflow time to be created
-    //         sleep($this->getConfig()->get('refresh_workflow_delay', 30));
-    //         $wfl = $wflc->fetch()->findLatestFromOptionsArray([
-    //             'target_commit' => $target_commit,
-    //         ]);
-    //         $current_time = time();
-    //         if ($end_time > 0 && $current_time >= $end_time) {
-    //             throw new TerminusException(
-    //                 'Exceeded maximum wait time of {max} seconds.',
-    //                 ['max' => $maxWaitInSeconds]
-    //             );
-    //         }
-    //     }
-
-    //     while (!$wfl->isFinished()) {
-    //         $current_time = time();
-    //         if ($end_time > 0 && $current_time >= $end_time) {
-    //             throw new TerminusException(
-    //                 'Exceeded maximum wait time of {max} seconds.',
-    //                 ['max' => $maxWaitInSeconds]
-    //             );
-    //         }
-    //         $this->log()->notice('Waiting for workflow {id} to complete.', ['id' => $wfl->id,]);
-    //         sleep($this->getConfig()->get('refresh_workflow_delay', 30));
-    //         $wfl->fetch();
-    //     }
-    //     $this->log()->notice('Workflow {id} has completed with status {status}.', [
-    //         'id' => $wfl->id,
-    //         'status' => $wfl->get('status'),
-    //     ]);
-    // }
+    protected function waitForCommit(
+        $startTime,
+        $site,
+        $env_name,
+        $target_commit,
+        $maxWaitInSeconds = 180
+    ) {
+        $current_time = time();
+        if ($maxWaitInSeconds > 0) {
+            $end_time = $current_time + $maxWaitInSeconds;
+        } else {
+            $end_time = 0;
+        }
+
+        // Validate commit SHA format
+        if (!preg_match('/^[0-9a-f]{40}$/', $target_commit)) {
```

Comment:
Should we really pass the whole sha? What about the common git/github thing of using first N chars?

Reply:
yes that's a good point.